website/static/js/wikiTreebeard.js

```diff
@@ -0,0 +1,246 @@
+function resolveToggle(item) {
+    var toggleMinus = m('i.fa.fa-minus', ' '),
+        togglePlus = m('i.fa.fa-plus', ' ');
+
+    if (item.children.length > 0) {
+        if (item.open) {
+            return toggleMinus;
+        }
+        return togglePlus;
+    }
+    item.open = true;
+    return '';
+}
+
+function expandOnLoad() {
+    var tb = this;  // jshint ignore: line
+    for (var i = 0; i < tb.treeData.children.length; i++) {
+        var parent = tb.treeData.children[i];
+        tb.updateFolder(null, parent);
+        expandChildren(tb, parent.children);
+    }
+}
+
+
+function expandChildren(tb, children) {
+    var openParent = false;
+    for (var i = 0; i < children.length; i++) {
+        var child = children[i];
+        var parent = children[i].parent();
+        if (child.children.length > 0) {
+            expandChildren(tb, child.children);
+        }
+    }
+    if (openParent) {
+        openAncestors(tb, children[0]);
+    }
+}
+
+function openAncestors (tb, item) {
+    var parent = item.parent();
+    if(parent && parent.id > 0) {
+        tb.updateFolder(null, parent);
+        openAncestors(tb, parent);
+    }
+}
+
+function before_change_permissions(item, permission){
+    var title = item.parent().data.node.title;
+    if(permission === 'public'){
+        bootbox.dialog({
+            title: 'Make publicly editable',
+            message: 'Are you sure you want to make the wiki of <b>' +title+
+                '</b> publicly editable? This will allow any logged in user to edit the content of this wiki. ' +
+                '<b>Note</b>: Users without write access will not be able to add, delete, or rename pages.',
+            buttons: {
+                cancel : {
+                    label : 'Cancel',
+                    className : 'btn-default',
+                    callback : function() {item.notify.update('', 'notify-primary', 1, 10);}
+                },
+                success: {
+                    label: 'Apply',
+                    className: 'btn-primary',
+                    callback: function() {change_permissions(item, permission);}
+                }
+            }
+        });
+    }
+    else {
+        change_permissions(item, permission);
+    }
+}
+
+function change_permissions(item, permission) {
+    var id = item.parent().data.node.id;
+
+    $osf.putJSON(
+        build_path(item), {'permission': permission}
+    ).done(function(){
+        item.notify.update('Settings updated', 'notify-success', 1, 2000);
+        item.data.select.permission = permission;
+    }).fail(function() {
+        item.notify.update('Could not update settings', 'notify-danger', 1, 2000);
+    });
+}
+
+// Helper to build path
+function build_path(item) {
+    var id = item.parent().data.node.id;
+    var permissions_change_path = '/api/v1/project/'+ id +
+        '/wiki/permissions/';
+    return permissions_change_path;
+}
+
+function ProjectWiki(data) {
```

Comment:
There is a lot of duplication with `notificationsTreebeard`. Consider how you might refactor this. 

If you decide that this can't be refactored without significant effort, add a TODO with your notes.